### PR TITLE
DATAMONGO-1404 DATAMONGO-1412 Add support for $min and $max update operators.

### PR DESCRIPTION
**Pull request by mp911de**
This PR is to review changes from #353 and back the changes by documenting rules for Java to MongoDB representation mapping

----
Related tickets: [DATAMONGO-1404](https://jira.spring.io/browse/DATAMONGO-1404) [DATAMONGO-1412](https://jira.spring.io/browse/DATAMONGO-1412)
Related pull request: #353

### CODE REVIEW

File: src/main/asciidoc/reference/mapping.adoc [hunk 1]
```diff
@@ -59,6 +59,186 @@ The following outlines what type conversion, if any, will be done on the propert
...
+
+This section explain how types are mapped to a MongoDB representation and vice versa. Spring Data MongoDB supports all types that can be represented as BSON, MongoDB's internal document format.
+In addition to these types, Spring Data MongoDB provides a set of built-in converters to map additional types. You can provide your own converters to adjust type conversion, see <<mapping-explicit-converters>> for further details.
+
+[cols="3,1,6", options="header"]
+.Type
+|===
+| Type
+| Type conversion
+| MongoDB representation
```
Comment (christophstrobl) [inline src/main/asciidoc/reference/mapping.adoc]: Should we rather call this _Sample_ as the actual representation of a `GeoJsonPoint` would just be `{ "type" : "Point" , "coordinates" : [3.0 , 4.0] }`.

File: src/main/asciidoc/reference/mapping.adoc [hunk 2]
```diff
@@ -59,6 +59,186 @@ The following outlines what type conversion, if any, will be done on the propert
...
+| converter
+| `{"polygon" : { "points" : [ { "x" : 1.0 , "y" : 2.0} , { "x" : 3.0 , "y" : 4.0} , { "x" : 4.0 , "y" : 5.0}]}}`
+
+| `Circle`
+| converter
+| `{"circle" : { "center" : { "x" : 1.0 , "y" : 2.0} , "radius" : 3.0 , "metric" : "NEUTRAL"}}`
+
+| `Point`
+| converter
+| `{ "x" : 1.0 , "y" : 2.0}`
```
Comment (christophstrobl) [inline src/main/asciidoc/reference/mapping.adoc]: we've `{"circle" : { "center" : {...` and `{"polygon" : { "points" : [ { "` so shouldn't we use `{"point" : { "x" : 1.0 , "y" : 2...` here?

Comment (mp911de) [inline src/main/asciidoc/reference/mapping.adoc]: Done.

File: src/main/asciidoc/reference/mapping.adoc (hunk 1 above)
Comment (mp911de) [inline src/main/asciidoc/reference/mapping.adoc]: Done.

**Pull request closed without merging**